modules/osd-monitoring-troubleshooting-issues.adoc

```diff
@@ -0,0 +1,109 @@
+----
++
+[NOTE]
+====
+You must run the command on a pod that has `curl` installed.
+====
++
+The following example output shows a result with a valid version metric.
++
+.Example output
+[source,terminal]
+----
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+# HELP version Version information about this binary-- --:--:-- --:--:--     0
+# TYPE version gauge
+version{version="v0.1.0"} 1
+100   102  100   102    0     0  51000      0 --:--:-- --:--:-- --:--:-- 51000
+----
++
+An invalid output indicates that there is a problem with the corresponding application.
+
+. If you are using PodMonitor, verify that PodMonitor is configured to point to the correct pods using label matching. For more information, see the Prometheus Operator documentation].
+. If you are using ServiceMonitor, and if the `/metrics` endpoint of the pod is showing metric data from step 2, follow these steps to verify the configuration:
+.. Run the following command to verify that the service is pointed to the correct `/metrics` endpoint. Ultimately, the service `labels` in this output must match the services monitor `labels` and the `/metrics` endpoint defined by the service in the subsequent steps.
++
+----
+$ oc get service
+----
++
+.Example output
+[source,terminal]
+----
+apiVersion: v1
+kind: Service <1>
+metadata:
+  labels: <2>
+    app: prometheus-example-app
+  name: prometheus-example-app
+  namespace: ns1
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+    targetPort: 8080
+    name: web
+  selector:
+    app: prometheus-example-app
+  type: ClusterIP
+----
++
+<1> Specifies that this is a service API.
+<2> Specifies the labels that are being used for this service.
+
+.. Query the `serviceIP`, `port`, and `/metrics` endpoints to see if the same metrics from the `curl` command you ran on the pod in step 2:
+... Run the following command to find the service IP:
++
+[source,terminal]
+----
+$ oc get service -n <target_namespace>
+----
+... Query the `/metrics` endpoint:
++
+[source,terminal]
+----
+$ oc exec <sample_pod> -n <sample_namespace> -- curl <service_IP>:<port>/metrics
+----
++
+Valid metrics are returned in the following example.
++
+.Example output
+[source,terminal]
+----
+% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                               Dload  Upload   Total   Spent    Left  Speed
+100   102  100   102    0     0  51000      0 --:--:-- --:--:-- --:--:--   99k
+# HELP version Version information about this binary
+# TYPE version gauge
+version{version="v0.1.0"} 1
+----
+.. Use label matching to verify that the `ServiceMonitor` object is configured to point to the desired service. To do this, compare the `Service` object from `oc get service` output to the `ServiceMonitor` object from `oc get servicemonitor` output. The labels should match.
```

Comment:
You could change "from `oc get service` output" to "from the `oc get service` output"

Reply:
Done.